tests/unit_tests/utils/test_model_data_writer.py

```diff
@@ -12,27 +13,29 @@
 logger.setLevel(logging.DEBUG)
 
 
-def test_write_metadata(args_dict_site, io_handler):
+def test_write_metadata(args_dict_site, io_handler, tmp_test_directory):
 
-    # test writer with no workflow configuration
-    w_1 = writer.ModelDataWriter()
-    with pytest.raises(
-        AttributeError, match=r"\'NoneType\' object has no attribute \'product_data_file_name\'"
-    ):
-        w_1.write_metadata()
+    # test writer of metadata
+    w_1 = writer.ModelDataWriter(args_dict=args_dict_site)
+    yml_file = w_1.write_metadata()
+    assert Path(yml_file).exists()
 
+    yml_file = w_1.write_metadata(ymlfile=str(tmp_test_directory) + "/test_file.yml")
+    assert Path(yml_file).exists()
 
-def test_write_data(args_dict_site, io_handler):
+    with pytest.raises(FileNotFoundError):
+        w_1.write_metadata(ymlfile="./this_directory_is_not_theta/test_file.yml")
 
-    # test writer with no workflow configuration
-    w_1 = writer.ModelDataWriter()
-    with pytest.raises(
-        AttributeError, match=r"\'NoneType\' object has no attribute \'product_data_file_name\'"
-    ):
-        w_1.write_data(None)
 
+def test_initialize():
+    # test writer with no workflow configuration and no configuration dictionary defined
+    with pytest.raises(TypeError, match=r"\'NoneType\' object is not subscriptable"):
```

Comment:
Is there a reason why here you use `r"\'` notation and below you use `re.escape`?

Reply:
Yes, I didn't know what `re.escape` is - not sure what the advantage is here? Should I?

Reply:
Either would be good, just being consistent is important (especially since they are in the same function).